client/src/pages/Plan/Plan.js

```diff
@@ -13,33 +13,16 @@ import { colors as c } from "../../colors";
 const reqsToolTip = "Requirements satisfied by the plan will be ticked off and become green";
 
 const Plan = () => {
-    const { student, realCourses, programme } = usePlan();
+    const { student, realCourses, programme, plan, courseAllocations, setCourseAllocations } = usePlan();
     const [searchTerm, setSearchTerm] = useState("");
     const [note, setNote] = useState("");
-    const start = 2020;
-    const end = 2021;
+    const start = plan.startYear;
+    const end = plan.endYear;
     const years = [];
-    for (var i = start; i <= end; i++) {
+    for (let i = start; i <= end; i++) {
         years.push(i);
     }
```

Comment:
Instead of these lines could we define a function at the top of the component:

`const generateYears = (from, to) => [...Array(to + 1).keys()].slice(from);`

and then in the component you can do `const years = generateYears(plan.startYear, plan.endYear);`

Reply:
You sure generating an array with 2000 elements is fast enough?

Reply:
![image](https://user-images.githubusercontent.com/39323194/93943508-c1609a00-fd86-11ea-82f2-5bb64dbbff0e.png)